### PR TITLE
Feat: Per Tenant Logging

### LINKED ISSUE #2359: Test Per Tenant Log Access and expand out
Follow up to PR#2234.

Basically test it out and apply it all applicable classes in ACA-Py.

### PR DESCRIPTION
**Pull request by shaangill025**
- resolve #2359 
- Derived from work done in #2425 
- Needed to close that PR as the pre-release images in there somehow included legacy code which was removed subsequently.

### CODE REVIEW

**Comment by shaangill025** [comment]
@WadeBarnes Can you please create a pre-release image for this PR.

**Comment by WadeBarnes** [comment]
@shaangill025, A prerelease image has been built; https://github.com/hyperledger/aries-cloudagent-python/pkgs/container/aries-cloudagent-python/138271403?tag=py3.9-0.11.0-pr2550-pre0

File: aries_cloudagent/admin/server.py [hunk 1]
```diff
@@ -407,6 +408,11 @@ async def setup_context(request: web.Request, handler):
                     ) = self.multitenant_manager.get_wallet_details_from_token(
                         token=token
                     )
+                    try:
+                        context_wallet_id.get()
+                    except LookupError:
+                        wallet_id = profile.settings.get("wallet.id")
+                        context_wallet_id.set(wallet_id)
```
Comment (dbluhm) [inline aries_cloudagent/admin/server.py:415]: Why not just set?

File: aries_cloudagent/protocols/coordinate_mediation/v1_0/handlers/tests/test_keylist_handler.py [hunk 2]
```diff
@@ -44,16 +46,20 @@ async def test_handler_no_active_connection(self, context):
...
-        assert "not acting as mediator" in caplog.text
-        assert "Keylist received: " not in caplog.text
+    async def test_handler_no_record(self, context):
+        with async_mock.patch.object(
+            test_module, "LOG", async_mock.MagicMock()
+        ) as mock_logger:
+            handler, responder = KeylistHandler(), MockResponder()
+            await handler.handle(context, responder)
+            assert mock_logger.warning.call_count == 1
+            assert mock_logger.info.call_count == 0
```
Comment (dbluhm) [inline aries_cloudagent/protocols/coordinate_mediation/v1_0/handlers/tests/test_keylist_handler.py:56]: Can you elaborate on why these changes are necessary?

Comment (dbluhm) [reply aries_cloudagent/protocols/coordinate_mediation/v1_0/handlers/tests/test_keylist_handler.py:56]: If the caplog fixture isn't behaving as expected, I think it would be better to define our own `caplog` fixture in `conftest.py` instead of touching all these tests.

**Comment by WadeBarnes** [comment]
@shaangill025, based on the testing, it looks like you'll need to update the logging to default to `${HOME}/log` (or `./log`), for example `/home/aries/log/` on the askar only images.  On OCP the non-privilege user can't write to `${HOME}`.

**Review by dbluhm — DISMISSED** [review]

**Comment by dbluhm** [comment]
Any word on when we can look at getting the failing tests resolved?

**Review by esune — APPROVED** [review]
I think changes look good, I am a bit confused about having same settings (e.g.: the log file path) in both `default_per_tenant_logging.ini` and `default_per_tenant_logging.yaml`, but I might be missing something.

@dbluhm if you are happy with the latest tweaks we can probably merge.

**Comment by shaangill025** [comment]
> I am a bit confused about having same settings (e.g.: the log file path) in both `default_per_tenant_logging.ini` and `default_per_tenant_logging.yaml`, but I might be missing something.

This is about supporting both `ini` (`logging FileConfig`) and `yml` (`logging DictConfig`) config files.

**Review by dbluhm — APPROVED** [review]

**Comment by swcurran** [comment]
Awesome — thanks, @shaangill025 .  Thanks for reviewing and merging @esune @dbluhm .

One more to go...
